public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -56,7 +56,16 @@ export function TraceView(props: Props) {
 
   const theme = useTheme();
   const traceTheme = useMemo(
-    () => ({ type: theme.isDark ? ThemeType.Dark : ThemeType.Light, servicesColorPalette: colors } as ThemeOptions),
+    () =>
+      ({
+        type: theme.isDark ? ThemeType.Dark : ThemeType.Light,
+        servicesColorPalette: colors,
+        components: {
+          TraceName: {
+            fontSize: theme.typography.size.lg,
+          },
+        },
+      } as ThemeOptions),
```

Comment:
When taking a final look at this just now, I noticed this.
Considering that `TracingView` doesn't rerender frequently (if it does, there's probably a bug?) and creating this theme object is a very fast operation, I would simply recreate it in a variable of type `ThemeOptions` on every render.
Improved readability and type checking (because obviously, one can provide excess props when using assertions) are the key points why I would use the approach I suggested but even in terms of performance, it actually might very well [also be faster not to use `useMemo` here](https://kentcdodds.com/blog/usememo-and-usecallback/).

Reply:
The thing is it can actually rerender a lot. The whole state is on this level like what row is collapsed or what section of the details is open and also the state of the zoom functionality. So having the theme non memoized would make all PureComponents down the line ineffective and mean lots of unnecessary rerendering.

Ideally the state would be done differently. In jaeger ui the state is propagated via redux and so it sidesteps lots of components but that was pulled out so that the components themselves can be reused.

Reply:
Good point about the effect on the pure components beneath it.
Well, that's it from me then. :)